pkg/kv/kvserver/store_split.go

```diff
@@ -163,7 +177,15 @@ func splitPreApply(
 		initClosedTS = &hlc.Timestamp{}
 	}
 	initClosedTS.Forward(r.GetCurrentClosedTimestamp(ctx))
-	if err := rsl.SetClosedTimestamp(ctx, stateRW, *initClosedTS); err != nil {
+
+	// Update the RHS applied state with the computed closed timestamp.
+	//
+	// TODO(pav-kv): pass in the whole struct instead of this long list of
+	// parameters which can accidentally miss some fields.
+	if err := rsl.SetRangeAppliedState(
```

Comment:
Now that we're at it, I'm realizing that this place is a footgun w.r.t. consistency checker. Since we're creating and marshalling the `RangeAppliedState` proto anew here, this has a risk that different replicas (on different CRDB versions) will persist different bytes. This might then trip the consistency checker.

Will add a TODO, we should do something better here with this CT computation. Best case: we don't need to overwrite `RangeAppliedState`.

Reply:
Though it's not just a problem in splits. It's with any commands. We solve this with below-raft migrations. So, false alarm.